pytest_html/plugin.py

```diff
@@ -157,24 +158,29 @@ def create_asset(self, content, extra_index,
         def append_extra_html(self, extra, extra_index, test_index):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                if self.self_contained:
+                content = extra.get('content')
+                if content.startswith(('file', 'http')) or \
+                        os.path.isfile(content):
+                    if self.self_contained:
+                        warnings.warn('Images added via file or link, '
+                                      'may not work as expected when '
+                                      'using --self-contained-html')
```

Comment:
I would prefer something like "Self-contained HTML report includes link to external resource: /foo/bar"